templates/zerver/help/saml-authentication.md

```diff
@@ -57,6 +57,57 @@ Zulip with various common SAML identity providers.
      * Optionally you can also send us an icon that should be shown on the button.
 1. We will take care of the server-side setup and let you know as soon as it's ready.
 
+## Configure SAML with Keycloak
+
+1. Make sure you have created your organization. We'll assume its URL is
+   `https://<subdomain>.zulipchat.com` in the instructions below.
+2. Make sure your Keycloak server is up and running. We assume the URL to your keycloak is
+   `https://keycloak.example.com` and your realm is `master` (which is the default realm)
+
+
+1. Export the **public certificate of your IdP**: \
+   Navigate to the realm settings, click on the tab `Keys` and then on the button `Certificate`. Copy the content and
+   save it as `{idp_name}.crt`:
+   ```
+   -----BEGIN CERTIFICATE-----
+   {Paste the content here}
+   -----END CERTIFICATE-----
+   ```
+2. Create a new client:
+    * Client-ID: `https://<subdomain>.zulipchat.com`
+    * Client Protocol: saml
+    * Client SAML Endpoint: leave this field empty
+3. Change some configurations of your keycloak client: \
+   Switch to the configurations tab `Settings`:
+    - Valid Redirect URIs: `https://<subdomain>.zulipchat.com/*`
+    - Base URL: `https://<subdomain>.zulipchat.com/complete/saml/`
+    - Client Signature Required: `Disable`
+4. Create attribute mappers:
+    * Switch to the configurations tab `Mappers`
+    * Create a mapper for firstname
+       * Property: `firstName`
+       * Friendly Name: `first_name`
+       * SAML Attribute Name: `first_name`
+       * SAML Attribute Name Format: `Basic`
+    * Create a mapper for lastname
+       * Property: `lastName`
+       * Friendly Name: `last_name`
+       * SAML Attribute Name: `last_name`
+       * SAML Attribute Name Format: `Basic`
+    * Create a mapper for email
+       * Property: `email`
+       * Friendly Name: `email`
+       * SAML Attribute Name: `email`
+       * SAML Attribute Name Format: `Basic`
+
+
+The Keycloak side of configuration should be ready! Send the following information to us at support@zulip.com:
+* The URL of your zulipchat-hosted organization
+* The URL of your SSO. It should resemble `https://keycloak.example.com/auth/realms/master` \
+  (Note: Replace `master` with your realm name)
+* The public certificate of your IdP from step 1 (`{idp_name}.crt`)
```

Comment:
Does keycloack not have an appropriate metadata endpoint from which these values could be grabbed by us (support@zulip.com) instead of requiring the customer to find and send these particular details?

Reply:
I have been searching for something like that but could not find one. Your keywords helped finding it. The endpoint is always ``https://keycloak.example.com/auth/realms/master/protocol/saml/descriptor`, where `master` is replaced with the realm name of the client.
That means, the user does not need to find the certificate and send it. I will change this section in the documentation. Where do you (the zulipchat.com team) keep track of this information?

Reply:
Great! Hmm, from our perspective we just have to configure these values on the server, so the process will be that the user sends us the metadata link via email, a person handling the support ticket (which currently will be me, given this is SAML :smile: ) opens the link, grabs the necessary values from the XML there and based on those tells Tim or Alex what exactly to add to our `settings.py`, the file in which to save the certificate etc.